aoc/src/main/scala/aoc2015/Day05.scala

```diff
@@ -0,0 +1,57 @@
+package aoc2015
+
+import scala.io.Source
+
+object Day05 extends App:
+
+  val lines: List[String] =
+    Source
+      .fromResource("aoc2015/Day05")
+      .getLines
+      .toList
+
+  def double(x: String): Boolean =
+    if x.slice(0, 1) == x.slice(1, 2) then true
+    else false
+
+  def triple(x: String): Boolean =
+    if x.slice(0, 1) == x.slice(2, 3) then true
+    else false
+
+  def containsThreeVowels(s: String): Boolean =
+    val vowels = Set('a', 'e', 'i', 'o', 'u')
+    val count = s.toList.count(vowels)
+    if count >= 3 then true
+    else false
+
+  def containsLetterTwice(s: String): Boolean =
+    if s.sliding(2).toList.map(x => double(x)).contains(true) then true
+    else false
+
+  def doesNotContainString(s: String): Boolean =
```

Comment:
naming - this method seems to be named _too_ generically

(im triggered by the validation on special cases `ab`, `cd`, `pq` and `xy`).

Reply:
ie. alternatively you can choose to _not_ name the individual validations, eg. in a `for` comprehension as a serie of if statements provides a nice syntax for that use-case like [here](https://github.com/nmcb/aoc2015/blob/main/src/main/scala/Day05.scala#L15-L20)